### PR TITLE
Add a local, buffering client for easy testing without mocking.

### PR DESCRIPTION
**Pull request by cory-stripe**
#### Summary
Adds a client that buffers locally for easy testing.

#### Motivation
With previous metrics libraries we've relied on mocking, which is a pain in the ass. It would be much better to locally accrue spans then use something like `select` to determine if the span in question was actually emitted.

#### Needs
This is pretty much a rubberstamp, but let me know if you think another storage mechanism would be preferable?

#### Test plan
Added a unit test!

r? @aditya-stripe

### CODE REVIEW

File: test/ssf/sample_test.rb [hunk 1]
```diff
@@ -75,6 +75,17 @@ def test_set_tag_with_nils
...
 
+    def test_local_buffer_send
+      s = Ssf::SSFSpan.new({
+        id: 123456,
+      })
+
+      c = SSF::LocalBufferingClient.new()
+      c.send_to_socket(Ssf::SSFSpan.encode(s))
+
+      assert_equal(1, c.buffer.length, 'We got 1 span')
```
Comment (redsn0w422) [inline test/ssf/sample_test.rb]: Shouldn't 'We got 1 span' tell us what went wrong?

File: lib/ssf/local_buffering_client.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  class LocalBufferingClient < Client
+
+    attr_reader :buffer
+
+    def initialize()
+      @buffer = []
+    end
+
+    def send_to_socket(message)
+      @buffer << message
```
Comment (aditya-stripe) [inline lib/ssf/local_buffering_client.rb]: This buffer never gets cleared. Is that intentional?

Comment (cory-stripe) [inline lib/ssf/local_buffering_client.rb]: Yup! That's the caller's problem. We could add a clear method for convenience though!

Comment (aditya-stripe) [inline lib/ssf/local_buffering_client.rb]: +1 on a `clear` method; at the very least it makes it clear that a new client should be initialized for every fixture in P-S

**Comment by aditya-stripe** [comment]
lgtm

**Pull request merged**